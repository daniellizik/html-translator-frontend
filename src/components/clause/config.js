export const QUERY = {
  targets: {
    NODE_NAME: {
      rules: ['EQUALS','NOT_EQUALS','LIKE','NOT_LIKE']
    },
    TEXT: {
      rules: ['EQUALS','NOT_EQUALS','LIKE','NOT_LIKE','REGEX']
    },
    ATTR_KEY: {
      rules: ['EQUALS','NOT_EQUALS','LIKE','NOT_LIKE','REGEX']
    },
    ATTR_VAL: {
      rules: ['EQUALS','NOT_EQUALS','LIKE','NOT_LIKE','REGEX']
    }
  }
}

export const MUTATION = {
  targets: {
    NODE_NAME: {
      rules: ['ALL_REPLACE']
    },
    TEXT: {
      rules: ['START_OF','END_OF','ALL_REPLACE','REGEX_REPLACE']
    },
    ATTR_KEY: {
      rules: ['START_OF','END_OF','REGEX_REPLACE','ALL_REPLACE','CONCAT']
    },
    ATTR_VAL: {
      rules: ['START_OF','END_OF','REGEX_REPLACE','ALL_REPLACE','CONCAT']
    }
  }
}

export const defaultQuery = {
  type: 'QUERY',
  target: 'TEXT',
  targetValue: '',
  rule: 'EQUALS',
  ruleValue: '',
  ruleValueFlags: ''
}

export const defaultMutation = {
  type: 'MUTATION',
  target: 'TEXT',
  targetValue: '',
  rule: 'START_OF',
  ruleValue: '',
  ruleValueFlags: ''
}

export const defaultClause = {
  name: '',
  minimized: false,
  active: false,
  rules: [defaultQuery],
  view: []
}