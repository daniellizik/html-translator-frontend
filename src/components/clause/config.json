{
  "query": {
    "targets": {
      "nodeName": {
        "rules": [
          "equals",
          "notEquals",
          "like",
          "notLike"
        ]
      },
      "text": {
        "rules": [
          "equals",
          "notEquals",
          "like",
          "notLike",
          "regex"
        ]
      },
      "attrKey": {
        "rules": [
          "equals",
          "notEquals",
          "like",
          "notLike",
          "regex"
        ]
      },
      "attrVal": {
        "rules": [
          "equals",
          "notEquals",
          "like",
          "notLike",
          "regex"
        ]
      }
    },
    "defaultQuery": {
      "type": "query",
      "target": "text",
      "targetValue": "",
      "rule": "like",
      "ruleValue": "",
      "ruleValueFlags": ""
    }
  },

  "mutator": {
    "targets": {
      "nodeName": {
        "rules": [
          "allReplace"
        ]
      },
      "text": {
        "rules": [
          "startOf",
          "endOf",
          "allReplace",
          "regexReplace"
        ]
      },
      "attrKey": {
        "rules": [
          "startOf",
          "endOf",
          "regexReplace",
          "allReplace",
          "add"
        ]
      },
      "attrVal": {
        "rules": [
          "startOf",
          "endOf",
          "regexReplace",
          "allReplace",
          "add"
        ]
      }
    },
    "defaultMutation": {
      "type": "mutation",
      "target": "text",
      "targetValue": "",
      "rule": "startOf",
      "ruleValue": "",
      "ruleValueFlags": ""
    }
  }

}


