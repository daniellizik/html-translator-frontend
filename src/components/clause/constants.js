export const CLAUSE_ADD = '@CLAUSE/CLAUSE_ADD'
export const CLAUSE_REMOVE = '@CLAUSE/CLAUSE_REMOVE'
export const CLAUSE_CHANGE_NAME = '@CLAUSE/CHANGE_NAME'

export const CLAUSE_INVALID_REGEXP = '@CLAUSE/CLAUSE_INVALID_REGEXP'
export const CLAUSE_INVALID_ATTRKEY = '@CLAUSE/CLAUSE_INVALID_ATTRKEY'

export const QUERY_ADD = '@CLAUSE/QUERY_ADD'
export const QUERY_REMOVE = '@CLAUSE/QUERY_REMOVE'

export const QUERY_CHANGE_TARGET = '@CLAUSE/QUERY_CHANGE_TARGET'
export const QUERY_CHANGE_TARGET_VALUE = '@CLAUSE/QUERY_CHANGE_TARGET_VALUE'
export const QUERY_CHANGE_RULE = '@CLAUSE/QUERY_CHANGE_RULE'
export const QUERY_CHANGE_RULE_VALUE = '@CLAUSE/QUERY_CHANGE_RULE_VALUE'
export const QUERY_CHANGE_RULE_VALUE_FLAGS = '@CLAUSE/QUERY_CHANGE_RULE_VALUE_FLAGS'

export const MUTATION_ADD = '@CLAUSE/MUTATION_ADD'
export const MUTATION_REMOVE = '@CLAUSE/MUTATION_REMOVE'

export const MUTATION_ADD_ATTR = '@CLAUSE/MUTATION_ADD_ATTR'
export const MUTATION_REMOVE_ATTR_BY_KEY = '@CLAUSE/MUTATION_REMOVE_ATTR_BY_KEY'
export const MUTATION_REMOVE_ATTR_BY_VALUE = '@CLAUSE/MUTATION_REMOVE_ATTR_BY_VALUE'
export const MUTATION_REMOVE_ALL_ATTRS = '@CLAUSE/MUTATION_REMOVE_ALL_ATTRS'

