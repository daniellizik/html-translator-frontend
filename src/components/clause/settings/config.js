export const targets = ['NODE_NAME', 'TEXT', 'ATTR_KEY', 'ATTR_VAL']
export const behaviors = ['APPLY_TO_CHILDREN', 'APPLY_TO_NODE']

export const QUERY = {
  targets: {
    NODE_NAME: {
      rules: [
        'EQUALS', 
        'NOT_EQUALS', 
        'LIKE', 
        'NOT_LIKE',
        'REGEX'
      ],
      behaviors: []
    },
    TEXT: {
      rules: [
        'EQUALS', 
        'NOT_EQUALS', 
        'LIKE', 
        'NOT_LIKE',
        'REGEX'
      ],
      behaviors: []
    },
    ATTR_KEY: {
      rules: [
        'EQUALS', 
        'NOT_EQUALS', 
        'LIKE', 
        'NOT_LIKE',
        'REGEX',
        'HAS_NONE',
        'HAS_AT_LEAST'
      ],
      behaviors: []
    },
    ATTR_VAL: {
      rules: [
        'EQUALS', 
        'NOT_EQUALS', 
        'LIKE', 
        'NOT_LIKE',
        'REGEX'
      ],
      behaviors: []
    }
  }
}

export const MUTATION = {
  targets: {
    NODE_NAME: {
      rules: ['ALL_REPLACE', 'START_OF','END_OF'],
      behaviors
    },
    TEXT: {
      rules: ['ALL_REPLACE', 'START_OF', 'END_OF', 'REGEX_REPLACE'],
      behaviors
    },
    ATTR_KEY: {
      rules: ['ALL_REPLACE', 'START_OF', 'END_OF', 'REGEX_REPLACE'],
      behaviors
    },
    ATTR_VAL: {
      rules: ['ALL_REPLACE', 'START_OF', 'END_OF', 'REGEX_REPLACE'],
      behaviors
    }
  }
}

// hardcoded for parse5 ast spec
export const targetMap = {
  TEXT: 'value',
  NODE_NAME: 'nodeName'
}

export const defaultQuery = {
  targetValue: '',
  rule: 'EQUALS',
  ruleValue: '',
  ruleValueFlags: ''
}

export const defaultMutation = {
  targetValue: '',
  rule: 'START_OF',
  ruleValue: '',
  ruleValueFlags: '',
  behavior: ''
}

export const defaultClause = {
  name: '',
  minimized: false,
  active: false,
  target: 'TEXT',
  type: 'CLAUSE',
  queries: [defaultQuery],
  mutations: [],
  view: []
}