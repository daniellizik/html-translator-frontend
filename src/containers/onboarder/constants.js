export const SKIP_ONBOARDING = '@ONBOARDER/SKIP_ONBOARDING'
export const ONBOARD_INIT = '@ONBOARDER/ONBOARD_INIT'
export const ONBOARD_5 = '@ONBOARDER/ONBOARD_5'