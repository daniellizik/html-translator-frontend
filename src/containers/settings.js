import React, { Component } from 'react'
import { connect } from 'react-redux'

class Settings extends Component {
  constructor() {
    super()
  }
  componentDidMount() {

  }
  render() {
    return !this.props.viewing ? null : (
      <div class="container-fluid max p-0 m-0">
        
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  viewing: state.user.viewingSettings
})

export default connect(mapStateToProps)(Settings)