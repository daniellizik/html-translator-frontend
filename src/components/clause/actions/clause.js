import * as constants from '../constants'

export default {
  activate: (clauseIndex) => ({
    type: constants.CLAUSE_ACTIVATE,
    clauseIndex
  }),
  add: () => ({
    type: constants.CLAUSE_ADD
  }),
  remove: (clauseIndex) => ({
    type: constants.CLAUSE_REMOVE,
    clauseIndex
  }),
  removeAll: () => ({
    type: constants.CLAUSE_REMOVE_ALL
  }),
  changeName: (clauseIndex, name) => ({
    type: constants.CLAUSE_CHANGE_NAME,
    clauseIndex,
    name
  }),
  changeTarget: (target, clauseIndex) => ({
    type: constants.CLAUSE_CHANGE_TARGET,
    target,
    clauseIndex
  }),
  setCurrentMutation: (currentMutation) => ({
    type: constants.CLAUSE_VIEW_MUTATIONS,
    currentMutation
  })
}