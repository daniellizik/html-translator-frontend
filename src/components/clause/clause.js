import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import policyValidator from '~/src/components/clause/policies/validator'
import { queryActions, mutateActions, clauseActions } from '~/src/components/clause/actions/index'
import * as config from '~/src/components/clause/settings/config'
import { ChangeTarget } from './clauses'
import { ToolTip, ChangeTargetExplanation } from '~/src/components/explanation'

const mapDispatchToProps = (dispatch) => ({
  queryActions: bindActionCreators(queryActions, dispatch),
  mutateActions: bindActionCreators(mutateActions, dispatch),
  clauseActions: bindActionCreators(clauseActions, dispatch)
})

const mapStateToProps = (state) => ({
  ...state,
  currentMutation: state.slave.currentMutation,
  onboardingStep: state.onboarding.step
})

export const Clause = connect(mapStateToProps, mapDispatchToProps)((props) => (
  <div class="py-0 bg-lightGrey">
    {policyValidator(props)}
    <div class="row py-2 m-0">
      <div class="col-4">
        <span onClick={() => props.actionSet[props.type].remove(props.clauseIndex, props.ruleIndex)}>
          {({QUERY: 'remove this query', MUTATION: 'remove this mutation'})[props.type]}
        </span>
      </div>
    </div>
  </div>
))

export const MaximizedClause = connect(mapStateToProps, mapDispatchToProps)(({ 
  currentMutation, 
  clauseActions, 
  queryActions, 
  mutateActions, 
  clauseIndex, 
  clauseGroup,
  onboardingStep
}) => (
  <div
    key={clauseIndex} 
    class="col-12 mx-0 mb-3 py-3 c-white bg-middleGrey">
    <div class="row px-3">
      <label class="col-6 mb-2 mx-0 pl-0 pr-2">
        <p>change target</p>
        <ToolTip
          placement="right"
          destroyTooltipOnHide={true}
          visible={onboardingStep === 4}
          trigger={clauseIndex === 0 ? ['hover'] : []}
          overlay={<ChangeTargetExplanation />}>
          <select 
            class="form-control custom-select"
            onFocus={() => clauseActions.activate(clauseIndex)}
            onChange={({target}) => clauseActions.changeTarget(target.value, clauseIndex)}>
            {config.targets.map((p, j) => (
              <option value={p} key={j}>
                {p}
              </option>
            ))}
          </select>
        </ToolTip>
      </label>

      <div class="col-12 mb-2 mx-0 px-0">
        <button class="btn mr-2 my-1" onClick={() => clauseActions.remove(clauseIndex)}>
          remove this clause
        </button>
        <button class="btn mr-2 my-1" onClick={() => clauseActions.denormalize(currentMutation === clauseIndex ? -1 : clauseIndex)}>
          {currentMutation === clauseIndex ? 'hide mutations' : 'view mutations'}
        </button>
        <button class="btn mr-2 my-1" onClick={() => queryActions.add(clauseIndex)}>
          add a query
        </button>
        <button class="btn mr-2 my-1" onClick={() => mutateActions.add(clauseIndex)}>
          add a mutation
        </button>
      </div>
      <div class="col-12 m-0 p-0">
        <div class="row m-0 p-0">
          <div class="col m-0 p-0">queries</div>
        </div>
        {clauseGroup.queries.map((clause, ruleIndex, {length}) => (
          <Clause 
            type="QUERY"
            actionSet={queryActions}
            clause={clause} 
            isLast={ruleIndex === length - 1} 
            clauseIndex={clauseIndex} 
            ruleIndex={ruleIndex} 
            key={`q-${clauseIndex}-${ruleIndex}`} />
        ))}
        <div class="row m-0 p-0">
          <div class="col m-0 p-0">mutations</div>
        </div>
        {clauseGroup.mutations.map((clause, ruleIndex, {length}) => (
          <Clause 
            type="MUTATION"
            actionSet={mutateActions}
            clause={clause} 
            isLast={ruleIndex === length - 1} 
            clauseIndex={clauseIndex} 
            ruleIndex={ruleIndex} 
            key={`m-${clauseIndex}-${ruleIndex}`} />
        ))}
      </div>
    </div>
  </div>
))

export const MinifiedClause = connect(mapStateToProps, mapDispatchToProps)((props) => (
  null
))