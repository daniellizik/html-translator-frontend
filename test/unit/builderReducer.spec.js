import { defaultMutation } from '~/src/components/clause/settings/config'
import * as constants from '~/src/components/clause/constants'
import { chainedDenormalizations } from '~/test/storyFixtures/mutation'

describe('reducing slave.view', () => {
  it('should generate slave.view for reduced clauses', () => {
    const { view } = chainedDenormalizations[13].slave
    expect(view.filter(n => !n.close && n.nodeName === 'span-good').length).toBe(5)
    expect(view.filter(n => !n.close && n.value === 'blah-foo').length).toBe(3)
  })
})