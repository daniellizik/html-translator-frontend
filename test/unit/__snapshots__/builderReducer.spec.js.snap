exports[`reducing slave.view should generate slave.view for reduced clauses 1`] = `
Object {
  "blah-foo": Array [],
  "span-good": Array [],
}
`;
